images/scripts/create_vm.sh

```diff
@@ -1,49 +1,45 @@
 #!/bin/bash
-qemu-img create -f qcow2 vdisk1 5G
-qemu-img create -f qcow2 vdisk2 5G
+set -eo pipefail
 
-ip link add link lan0 name macvtap0 type macvtap mode passthru
-ip link set macvtap0 up
-ifconfig macvtap0 promisc
+if [ "$1" == "e0ab02d2-27cd-5a5e-8efc-080ba80cf258" ]; then
```

Comment:
That's probably the best way to go at the moment with this script). I also was working on parametrizing it. Tried to use `for` loop in this script: https://gist.github.com/GrigoriyMikhalkin/f578adbf453fe5b7643d71fdd947e1cc -- but `net` option always fails at VM set up. Documentation on that part is very poor and it even looks like `net` option is legacy at the moment.

I think, right now it is best to use your variant. And later do research on `netdev` option, to replace `net` with it.

One little suggestion, let's use associative array that points machine name to UUID. So that script could be started like `create_vms machine01` instead of passing UUID. IMO, that's more developer friendly interface.

Reply:
Sounds all reasonable. You can push your array improvement on this branch if you want.

Reply:
I was just able to successfully use my script with minor changes) Push it in a few minutes. What you think, should we use it instead?

Reply:
Working version of the script: https://github.com/metal-stack/mini-lab/blob/010ad85b867a554a1f73c05ce7dcec5ed1152c93/images/scripts/create_vms.sh

Reply:
I am not so good with bash to be honest. Can't judge if it works or not but it looks kind of difficult. Maybe we should do this in another programming language?

Reply:
Writing it in a real programming language would easy further development a lot, i doubt that in the future someone is keen enough to modify this file

What about go/python ?

Reply:
I tried in Python but due to the Bash magic going on with `3<>/dev/tap$(cat /sys/class/net/macvtap0/ifindex)` I was not able to make it. :(

Reply:
It's now done in Python but executed directly through Bash with `shell: True`.